modules/quest/src/main/java/unnamed/mmo/quest/objective/QuestObjective.java

```diff
@@ -0,0 +1,16 @@
+package unnamed.mmo.quest.objective;
+
+import net.minestom.server.entity.LivingEntity;
+import net.minestom.server.instance.block.Block;
+import unnamed.mmo.quest.QuestProgress;
+
+public interface QuestObjective {
+
+    default void onMobKill(LivingEntity entity) {}
```

Comment:
Adding methods for all of these is going to get extremely cumbersome, probably would be better to expose an event node or something here that the objective can register listeners to on its own